sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/BinaryData.java

```diff
@@ -43,19 +53,32 @@
  * <p><strong>Get an Object from {@link BinaryData}</strong></p>
  * {@codesnippet com.azure.core.experimental.util.BinaryDocument.to#ObjectAsync}
  *
+ * <p><strong>Create an instance from Object</strong></p>
+ * {@codesnippet com.azure.core.experimental.util.BinaryDocument.from#Object}
+ *
  * @see ObjectSerializer
  */
 public final class  BinaryData {
     private static final ClientLogger LOGGER = new ClientLogger(BinaryData.class);
+    private static final byte[] EMPTY_BYTES = new byte[0];
+    private static final BinaryData EMPTY_DATA = new BinaryData(new byte[0]);
+
+    private static final Object LOCK = new Object();
+
     private final byte[] data;
 
+    private static volatile JsonSerializer defaultJsonSerializer;
```

Comment:
This should not be static

Reply:
CheckStyle complain if I change to this  -> `Name 'DEFAULT_JSON_SERIALIZER' must match pattern '^[a-z][a-zA-Z0-9]*$'. [StaticVariableName]`